[sdl2] build fails when VCPKG_BUILD_TYPE is set to "release"

**Host Environment**
- OS: Any
- Compiler: Any

**To Reproduce**
Steps to reproduce the behavior:
1. Define `VCPKG_BUILD_TYPE` to `release` (i.e.. `VCPKG_BUILD_TYPE=release`)
2. `./vcpkg install sdl2`

**Failure logs** 
```
Building package sdl2[core,vulkan]:x86-windows...
[...]
-- Configuring x86-windows
-- Building x86-windows-rel
CMake Error at scripts/cmake/vcpkg_replace_string.cmake:11 (file):
  file failed to open for reading (No such file or directory):

    D:/[...]/build/vcpkg/packages/sdl2_x86-windows/debug/lib/pkgconfig/sdl2.pc
Call Stack (most recent call first):
  ports/sdl2/portfile.cmake:95 (vcpkg_replace_string)
  scripts/ports.cmake:79 (include)
```

**Additional context**
PR forthcoming with fix

--- a/ports/sdl2/CONTROL
+++ b/ports/sdl2/CONTROL
@@ -1,9 +1,0 @@
-Source: sdl2
-Version: 2.0.12
-Port-Version: 5
-Homepage: https://www.libsdl.org/download-2.0.php
-Description: Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.
-
-Feature: vulkan
-Description: Vulkan functionality for SDL
-Build-Depends: vulkan

--- a/ports/sdl2/CONTROL
+++ b/ports/sdl2/CONTROL
@@ -1,6 +1,6 @@
 Source: sdl2
 Version: 2.0.12
-Port-Version: 4
+Port-Version: 5
 Homepage: https://www.libsdl.org/download-2.0.php
 Description: Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.
 

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -92,7 +92,9 @@ file(STRINGS "${SOURCE_PATH}/CMakeLists.txt" DYLIB_CURRENT_VERSION REGEX ${DYLIB
 string(REGEX REPLACE ${DYLIB_COMPATIBILITY_VERSION_REGEX} "\\1" DYLIB_COMPATIBILITY_VERSION "${DYLIB_COMPATIBILITY_VERSION}")
 string(REGEX REPLACE ${DYLIB_CURRENT_VERSION_REGEX} "\\1" DYLIB_CURRENT_VERSION "${DYLIB_CURRENT_VERSION}")
 
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2" "-lSDL2d")
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+	vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/sdl2.pc" "-lSDL2" "-lSDL2d")
+endif()
 
 vcpkg_fixup_pkgconfig(
     IGNORE_FLAGS "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/lib/pkgconfig/../../lib" "-Wl,-rpath,${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/../../lib" "-Wl,--enable-new-dtags" "-Wl,--no-undefined" "-Wl,-undefined,error" "-Wl,-compatibility_version,${DYLIB_COMPATIBILITY_VERSION}" "-Wl,-current_version,${DYLIB_CURRENT_VERSION}" "-Wl,-weak_framework,Metal" "-Wl,-weak_framework,QuartzCore"

--- a/ports/sdl2/vcpkg.json
+++ b/ports/sdl2/vcpkg.json
@@ -1,0 +1,15 @@
+{
+  "name": "sdl2",
+  "version-string": "2.0.12",
+  "port-version": 5,
+  "description": "Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.",
+  "homepage": "https://www.libsdl.org/download-2.0.php",
+  "features": {
+    "vulkan": {
+      "description": "Vulkan functionality for SDL",
+      "dependencies": [
+        "vulkan"
+      ]
+    }
+  }
+}

